src/renderer/services/neo.js

```diff
@@ -789,6 +964,107 @@ export default {
+    let net;
+
+    if (currentNetwork.net === 'MainNet') {
+      net = MAINNET_BTC;
+    } else {
+      net = TESTNET_BTC;
+    }
+
+    return new Promise(async (resolve, reject) => {
+      try {
+        store.commit('setSendInProgress', true);
+        try {
+          bitcoin.address.toOutputScript(toAddress, net);
+        } catch (e) {
+          return reject('Invalid Recipient Address.');
+        }
+
+        const pendingTxProposals = await valuation.getPendingTxProposal(currentWallet.btcWalletClient);
+
+        pendingTxProposals.forEach(async (pendingTxProposal) => {
+          try {
+            const rmtx = await valuation.removeTxProposal(pendingTxProposal);
+            console.log(rmtx);
+          } catch (e) {
+            reject('error in removing pending transaction proposals!');
+          }
+        });
+
+        const balanceTemp = await valuation.getBTCBalanceByAddress(currentWallet.btcWalletClient);
+        const balanceBTC = balanceTemp.availableConfirmedAmount / 100000000;
+        const toAmountBTC = amount;
+        const toAmountSAT = parseInt((amount * 100000000).toFixed(0), 10);
+        let output1;
+        let output2;
+        let output3;
+        let output4;
+        const txp = {
+          outputs: [{
+            toAddress,
+            amount: toAmountSAT,
+          }],
+          feeLevel: 'normal',
+          excludeUnconfirmedUtxos: true,
+          dryRun: false,
+        };
+        try {
+          output1 = await valuation.createTxProposal(currentWallet.btcWalletClient, txp);
+        } catch (e) {
+          return reject(e.err.message);
+        }
+        const feeBTC = parseInt((output1.createdTxp.fee * 100000000).toFixed(0), 10);
+
+        const finalAmountBTC = toAmountBTC + feeBTC;
+
+        if (finalAmountBTC > balanceBTC) {
+          return reject(`Insufficiant BTC! Need ${finalAmountBTC} but only found ${balanceBTC}`);
+        }
+
+        try {
+          output2 = await valuation.publishTxProposal(currentWallet.btcWalletClient, output1.createdTxp);
+        } catch (e) {
+          return reject('Something went wrong while publishing transaction proposal!');
+        }
+
+        try {
+          /* eslint-disable max-len */
+          output3 = await valuation.signTxProposal(currentWallet.btcWalletClient, output2.publishedTxp, currentWallet.btcKey);
+          /* eslint-enable max-len */
+        } catch (e) {
+          try {
+            const rm3 = await valuation.removeTxProposal(currentWallet.btcWalletClient, output2.publishedTxp);
+            console.log(rm3);
+          } catch (err2) {
+            return reject('Something went wrong while signing transaction proposal!');
+          }
+          return reject('Something went wrong while signing transaction proposal!');
+        }
+
+        try {
+          output4 = await valuation.broadcastTxProposal(currentWallet.btcWalletClient, output3.signedTxp);
+        } catch (e) {
```

Comment:
We can't do it if the transaction proposal already published then it should be completed or rejected otherwise the wallet will stuck, so if the transaction does not sends due to a network issue, it will not retries, and on the next transaction, all pending transaction proposals will be removed.